src/main/java/training/endava/skeleton/MainApplication.java

```diff
@@ -1,64 +1,223 @@
+        Person p111 = new Person(1, "Gigel1", "07XXXXXXX1", "Iasi");
+        Person p211 = new Person(2, "Gigel2", "07XXXXXXX1", "Botosani");
+        Person p311 = new Person(3, "Gigel3", "07XXXXXXX1", "Suceava");
 
-        System.out.println("-----------------------------------------------------------------------------------------");
-//        pr.deleteById(9);
-//        System.out.println(persons);
+        persons1.addAll(Arrays.asList(p111, p211, p311));
+
+
+        List<Company> companiesList = new ArrayList<>();
+        Company c1 = new Company("Company 1", 2010, persons, CompanyArea.Finance, 1000000);
+        Company c2 = new Company("Company 2", 2011, persons1, CompanyArea.CustomerSupport, 200000);
+        Company c3 = new Company("Company 3", 2012, persons2, CompanyArea.It, 500000);
+
+        companiesList.addAll(Arrays.asList(c1, c2, c3));
+
+
+
+
+        System.out.println("****************************************************************************");
+        //1. List the names of all companies.
+
+        List<String> namesOfAllCompanies = companiesList.stream()
+                .map(p -> p.getName())
+                .distinct()
+                .collect(Collectors.toList());
+
+        System.out.println(namesOfAllCompanies);
+
+
+        System.out.println("****************************************************************************");
+
+
+        // 2. Transform one of the object's property into a primitive stream and do some processing on it.
+
+        companiesList.stream()
+                .mapToDouble(Company::getProfit)
+//                .average()
+                .max()
+                .ifPresent(System.out::println);
+
+
+        System.out.println("****************************************************************************");
+        // 3. Find the oldest company based on the foundation year.
+
+        Company oldestCompanyBasedOnFoundationYear = companiesList.stream()
+                .min(Comparator.comparing(Company::getYear)).get();
+
+        System.out.println(oldestCompanyBasedOnFoundationYear);
+
+
+
+        System.out.println("****************************************************************************");
+        //     4. Count the persons in the stream using map() and reduce().
+
+        int nrOfPersons = companiesList.stream()
+                .map(Company::getPersonList)
+                .map(List::size)
+                .reduce(0, (t1, t2) -> t1 + t2);
+
+        System.out.println(nrOfPersons);
+
+
+
+        System.out.println("****************************************************************************");
+        //5. Use reduce() to create a Person instance that has the name created as a concatenation of all the other persons' names.
+
+        companiesList.stream()
+                .map(Company::getPersonList)
+                .flatMap(List<Person>::stream)
+                .reduce(new Person(),(per1,per2)->{per1.setName(per1.getName()+per2.getName()); return per1;});
```

Comment:
Have you verified the resulted name? per1.getName() for the identity value will be null, so the resulted person will have the name starting with "null". You should initialize the name with empty string in the constructor.